### ISSUE TITLE
Improvements in activity development tutorial

### ISSUE THREAD

**Issue by S-kus** [llaske/sugarizer#996]
There are some mistakes in the activity development tutorial:
- In [Sugarizer Vue.js activity development tutorial step-5 ](https://github.com/llaske/sugarizer/blob/dev/docs/tutorial/VueJS/step5.md) `localize` method for localizing the `l10n` object should be called in mounted, not in `localized` function. Because of this only backdrop is showing for the tutorial tour as no text is passing also `stringAddPawn` is not rendering. As shown:

![tour](https://user-images.githubusercontent.com/86892991/157531386-e04bc72c-4487-4db5-90a2-3a24fba3b5a9.png)

If I used the same mounted function used in the `SugarTutorial.js` component, then it's working perfectly fine.

- Given, `go-left.svg` is not correct in the Vue.js template. Because of this in tour `div` it's not rendering clearly. As shown:

![arrow](https://user-images.githubusercontent.com/86892991/157531058-50f19a0c-ca5f-474a-9ca0-05f8c5e5c445.png)

![go-left](https://user-images.githubusercontent.com/86892991/157531075-e3924bd8-8e24-4824-ba51-2c6eeaf5d412.png)

- In [Sugarizer Vanilla Javascript activity development tutorial step-1](https://github.com/llaske/sugarizer/blob/dev/docs/tutorial/VanillaJS/step1.md) the example image contains "locale.ini" file. It's not there in the empty [template provided](https://github.com/llaske/sugarizer/tree/master/activities/ActivityTemplate/VanillaJS). So, updating the image would be better for the newbie.

- There are minor spelling mistakes in some step instructions.

**Comment by S-kus** [comment]
Hey @llaske, I found these issues while implementing [Sugarizer Vanilla Javascript activity development tutorial](https://github.com/llaske/sugarizer/blob/dev/docs/tutorial/VanillaJS/tutorial.md) and the [Sugarizer Vue.js activity development tutorial](https://github.com/llaske/sugarizer/blob/dev/docs/tutorial/VueJS/tutorial.md). 
Could you please review the pull request? I will do the required correction if needed.

**Comment by llaske** [comment]
@S-kus Thanks for these feedbacks.

1) No. The right place for the call is in localized method. Here's the code:

	    mounted: function () {
		  this.SugarL10n = this.$refs.SugarL10n;
		  this.SugarPresence = this.$refs.SugarPresence;
	    },
	    methods: {
		initialized: function () {
			// Initialize Sugarizer
			this.currentenv = this.$refs.SugarActivity.getEnvironment();
			this.$refs.SugarL10n.$on('localized', this.localized());
		},

		localized: function () {
			this.displayText = this.SugarL10n.get("Hello", { name: this.currentenv.user.name });
			this.SugarL10n.localize(this.l10n);
		},

2) The render is right on my side:

<img width="313" alt="image" src="https://user-images.githubusercontent.com/981841/157971866-cc72536c-fe3a-45a5-b0dd-adb2eaf674f5.png">

3) The image doesn't need to match exactly the real content.

4) Okay. Nice.

**Comment by S-kus** [comment]
> 1. No. The right place for the call is in localized method. Here's the code:

Actually, I followed the tutorial and created the step vise commit in [this repository](https://github.com/S-kus/Pawn-Activity-Vue.js).
When I added the last commit changes i.e. [this one](https://github.com/S-kus/Pawn-Activity-Vue.js/commit/dff67d40f165f7b6467543acc477e2bd8cb06fc7) after the mentioned issue is fixed! Is there any codebase of this tutorial from where I can match and find my bug because I go through the tutorial again but didn't find any mistake.
For now, should I correct the commit in the pull request?




> 2\. The render is right on my side:

Could you please confirm the hex code of this `go-left.svg` in your code. As the code given in the template is using [the one with colored one](https://github.com/llaske/sugarizer/blob/dev/activities/ActivityTemplate/VueJS/icons/go-left.svg) but for `go-right.svg` [it's transparent type](https://github.com/llaske/sugarizer/blob/dev/activities/ActivityTemplate/VueJS/icons/go-right.svg). 




> 3\. The image doesn't need to match exactly the real content.

Oh, I will correct this commit.

**Comment by llaske** [comment]
> where I can match and find my bug because I go through the tutorial again but didn't find any mistake.

There is not a tutorial correction but I can send it to you if you want to explore the problem on your side. Send me a mail if you're interested.

> For now, should I correct the commit in the pull request?

Nope. In my opinion it's not an issue.

> Could you please confirm the hex code of this go-left.svg in your code.

        <!ENTITY stroke_color "#010101">
        <!ENTITY fill_color "#FFFFFF">

**Comment by S-kus** [comment]
> There is not a tutorial correction but I can send it to you if you want to explore the problem on your side. Send me a mail if you're interested.

Yes, Thank you @llaske. My email is **saumyakus@gmail.com**

> Nope. In my opinion, it's not an issue.

Okay, I will remove its correction commit from the pull request.



>     <!ENTITY stroke_color "#010101">
>     <!ENTITY fill_color "#FFFFFF">

Yes, exactly what I want to say! In the given VueJs template the [hex code given is different](https://github.com/llaske/sugarizer/blob/dev/activities/ActivityTemplate/VueJS/icons/go-left.svg?short_path=12942f1#:~:text=%3C!ENTITY%20stroke_color%20%22%23FFFFFF,ENTITY%20fill_color%20%22%23808080%22%3E) and there is no instruction to change it, So it's better if we attach the correct SVG code.

**Comment by llaske** [comment]
> Yes, Thank you @llaske. My email is [saumyakus@gmail.com](mailto:saumyakus@gmail.com)

I've sent it to you.

> Yes, exactly what I want to say! In the given VueJs template the [hex code given is different](https://github.com/llaske/sugarizer/blob/dev/activities/ActivityTemplate/VueJS/icons/go-left.svg?short_path=12942f1#:~:text=%3C!ENTITY%20stroke_color%20%22%23FFFFFF,ENTITY%20fill_color%20%22%23808080%22%3E) and there is no instruction to change it, So it's better if we attach the correct SVG code.

These lines are use only when icons are drawn using [Sugar drawing method](https://github.com/llaske/svgstudy#how-coloured-svg-file-works-in-sugarizer-v1) to recolorize it. It's not the case for tutorial buttons. So I don't see the interest to update it.

**Comment by S-kus** [comment]
> I've sent it to you.

Yeah got it. Thank you.



> These lines are use only when icons are drawn using [Sugar drawing method](https://github.com/llaske/svgstudy#how-coloured-svg-file-works-in-sugarizer-v1) to recolorize it. It's not the case for tutorial buttons. So I don't see the interest to update it.

Okay, sure I corrected the code for this change too in the pull request. I guess it can be merged now.

**Comment by S-kus** [comment]
Thanks, @llaske, for sharing your code I found the problem in my codebase!
In **activity.js file**

in your code `methods{}`: 

```
initialized: function () { 
     // Initialize Sugarizer 
        this.currentenv = this.$refs.SugarActivity.getEnvironment(); 
        this.$refs.SugarL10n.$on('localized', this.localized()); 
},
```

But in my code methods: 

```
initialized: function () { 
     // Initialize Sugarizer 
     this.currentenv = this.$refs.SugarActivity.getEnvironment(); 
},
```
i.e. 'localized' method is not called that's why when I called it during mounted it's worked. 
In the tutorial, it's not mentioned to add this `this.$refs.SugarL10n.$on('localized', this.localized());` in `initialized function` at any step. I guess it's should be corrected in the tutorial docs.

**Comment by llaske** [comment]
> i.e. 'localized' method is not called that's why when I called it during mounted it's worked.

Hmmm. I think you don't need to do call the method if you've inserted the `v-on` directive in the HTML. It does the same thing.

     <sugar-localization ref="SugarL10n" v-on:localized="localized"></sugar-localization>

**Comment by S-kus** [comment]
> Hmmm. I think you don't need to do call the method if you've inserted the `v-on` directive in the HTML. It does the same thing.

Yeah, you are right. Sorry for inconvenience, I tested things again in my code by using `console.log()` in functions and find out in my `index.html` I had written this `<sugar-activity ref="SugarActivity" v-on:initialized="initialized"></sugar-activity>` two times. Because of which my things are not working properly. I removed the 2nd instance of this code and it's working fine.

Thanks again. I guess then we can close this issue.

**Comment by llaske** [comment]
Nice to hear that you find the problem. It's a good way to learn.

**Comment by S-kus** [comment]
> Nice to hear that you find the problem. It's a good way to learn.

Thank You @llaske. Yes, it's helped a lot.
